CentralComputing/Sensor_Test.cpp

```diff
@@ -0,0 +1,388 @@
+	} else if (r == 3) {	
+		cout << "http://www.mememaker.net/static/images/memes/4284241.jpg" << endl;
+	} else if (r == 4) {
+		cout << "https://cdn.meme.am/cache/instances/folder695/63313695.jpg" << endl;
+	}
+	cout << endl;
+}
+
+/**
+	checks if the current state of pod matches witht the check state command
+	word: check state command to be executed
+	pod: reference of pod to be checked
+
+	return: whether the command returned true or false
+*/
+bool checkState(string word, Pod& pod) {
+	size_t start = word.find("[");
+	size_t end = word.find("]");
+	string arg = word.substr(start+1,end-1);
+	int state = pod.getCurrentState();
+
+	// pos handles if (current_state == some_state) should return true or false
+	bool pos = true;
+	if(arg.front() == '!') {
+		// negates the positive claim
+		pos = false;
+		// removes ! from argument
+		arg = arg.substr(1);
+	}
+	
+	// switch case for all the different states
+	// uses
+	switch(state) {
+		case Pod::ST_SAFE_MODE:
+			// if the arg matches for this case
+			if(arg == "SAFE_MODE" || arg == "S_M") {
+				// return whatever pos is
+				return pos;
+			}
+			// other wise return the negate of pos
+			return !pos;
+			break;
+		case Pod::ST_FUNCTIONAL_TEST:
+			if(arg == "FUNCTIONAL_TESTS" || arg == "F_T") {
+				return pos;
+			}
+			return !pos;
+			break;
+		case Pod::ST_FLIGHT_ACCEL:
+			if(arg == "FLIGHT_ACCEL" || arg == "F_A") {
+				return pos;
+			}
+			return !pos;
+			break;
+		case Pod::ST_FLIGHT_COAST:
+			if(arg == "FLIGHT_COAST" || arg == "F_C") {
+				return pos;
+			}
+			return !pos;
+			break;
+		case Pod::ST_FLIGHT_BRAKE:
+			if(arg == "FLIGHT_BRAKE" || arg == "F_B") {
+				return pos;
+			}
+			return !pos;
+			break;
+		default:
+			return false;
+			break;
+	}
+}
+
+/**
+	Processes a line of code and executes on the pod object
+	word: the line of code to execute
+	pod: a reference to the pod object to be used
+
+	returns: a bool for the success of the process
+		Usually true
+*/
+bool process( string word, Pod& pod ) {
+	std::transform(word.begin(), word.end(), word.begin(), ::toupper);
+	// TODO add other events like sensor readings, etc;
+	
+	// checks for basic fsm commands
+	if (word == "MOVE_FUNCTIONAL_TESTS" || word == "MV_F_T") {
+		pod.move_functional_tests();
+	}
+	else if ( word == "MOVE_SAFE_MODE" || word == "MV_S_M") {
+		pod.move_safe_mode();
+	}
+	else if ( word == "MOVE_FLIGHT_ACCELERATION" || word == "MV_F_A") {
+		pod.move_flight_acceleration();
+	}
+	else if ( word == "EMERGENCY_BRAKE" || word == "E_BRAKE" ) {
+		pod.emergency_brake();
+	}
+	else if ( word == "COAST" || word == "C" ) {
+		pod.coast();
+	}
+	else if ( word == "BRAKE" || word == "B" ) {
+		pod.brake();
+	}
+	// checks for utility commands
+	else if ( word == "PRINT_STATE" || word == "PS" ) {
+		cout << pod.get_current_state_string() << endl;
+	}
+	else if ( word.substr(0,1) == "[") {
+		bool result = checkState(word, pod);
```

Comment:
I suggest:
`if( !check_state(word, pod) ){
      ....
      return false;
}` which means we preform the `if` when there is an error. 
Otherwise, proceed as usual. Only return false inside that if